### PR TITLE
Add smarty3 dirs config

### PR DESCRIPTION
**Pull request by fabrixxm**
`FriendicaSmarty` gets work dirs from config, with fallback if not set, and write permission check in `FriendicaSmartyEngine` report which folders aren't writable

### CODE REVIEW

File: src/Render/FriendicaSmarty.php [hunk 1]
```diff
@@ -47,11 +48,16 @@ function __construct(string $theme, array $theme_info)
...
 		$this->setTemplateDir($template_dirs);
 
-		$this->setCompileDir('view/smarty3/compiled/');
-		$this->setConfigDir('view/smarty3/');
-		$this->setCacheDir('view/smarty3/');
 
-		$this->left_delimiter = Renderer::getTemplateLeftDelimiter('smarty3');
+		$compile_dir = DI::config()->get('smarty3', 'compile_dir') ?? DI::basePath() . '/view/smarty3/compiled/';
+		$config_dir  = DI::config()->get('smarty3', 'config_dir')  ?? DI::basePath() . '/view/smarty3/';
+		$cache_dir   = DI::config()->get('smarty3', 'cache_dir')   ?? DI::basePath() . '/view/smarty3/';
```
Comment (MrPetovan) [inline src/Render/FriendicaSmarty.php]: `Config->get` has a default value parameter we can use here.
```suggestion
		$compile_dir = DI::config()->get('smarty3', 'compile_dir', DI::basePath() . '/view/smarty3/compiled/');
		$config_dir  = DI::config()->get('smarty3', 'config_dir', DI::basePath() . '/view/smarty3/');
		$cache_dir   = DI::config()->get('smarty3', 'cache_dir', DI::basePath() . '/view/smarty3/');
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by MrPetovan** [comment]
Several checks are failing, are you sure you branched off the current `develop` ? If not, please rebase your branch on top of it.

**Comment by fabrixxm** [comment]
rebased over last `develop` and changed to use default value parameter

**Comment by MrPetovan** [comment]
Ah, you added a translation string so you need to run `bin/run_xgettext.sh` and commit the result to clear one of the checks.

**Comment by MrPetovan** [comment]
Also some API tests are now throwing a Smarty exception? Might want to look into this:
```
2) Friendica\Test\src\Module\Api\Twitter\FavoritesTest::testApiFavorites
Friendica\Network\HTTPException\ServiceUnavailableException: Friendica can't display this page at the moment, please contact the administrator.

/woodpecker/src/github.com/friendica/friendica/src/Render/FriendicaSmartyEngine.php:84
/woodpecker/src/github.com/friendica/friendica/src/Core/Renderer.php:169
/woodpecker/src/github.com/friendica/friendica/src/Core/Renderer.php:110
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:1207
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2078
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:1097
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:1077
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2076
/woodpecker/src/github.com/friendica/friendica/src/Util/Strings.php:497
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2398
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2085
/woodpecker/src/github.com/friendica/friendica/src/Util/Strings.php:497
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2398
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:2098
/woodpecker/src/github.com/friendica/friendica/src/Content/Text/BBCode.php:1463
/woodpecker/src/github.com/friendica/friendica/src/Factory/Api/Twitter/Status.php:143
/woodpecker/src/github.com/friendica/friendica/src/Factory/Api/Twitter/Status.php:109
/woodpecker/src/github.com/friendica/friendica/src/Module/Api/Twitter/Favorites.php:70
/woodpecker/src/github.com/friendica/friendica/src/BaseModule.php:238
/woodpecker/src/github.com/friendica/friendica/src/Module/BaseApi.php:99
/woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Twitter/FavoritesTest.php:41
```

**Comment by MrPetovan** [comment]
There seem to still be an issue with the translation file.

File: src/Render/FriendicaSmarty.php [hunk 2]
```diff
@@ -22,6 +22,7 @@
 namespace Friendica\Render;
 
 use Smarty;
+use Friendica\DI;
```
Comment (nupplaphil) [inline src/Render/FriendicaSmarty.php]: Fortunately, this is a class with a constructor. Please add the arguments at `__constructor()` and directly use the Dependency injection

Comment (nupplaphil) [reply src/Render/FriendicaSmarty.php]: Or even better add `string $workdir` at the constructor and avoid a dependency onto a complex structure/class by delegate it to the caller :-)

**Review by nupplaphil — CHANGES_REQUESTED** [review]

**Comment by fabrixxm** [comment]
> There seem to still be an issue with the translation file.

I see, but I can't tell where the problem is

File: src/Render/FriendicaSmarty.php (hunk 2 above)
Comment (fabrixxm) [reply src/Render/FriendicaSmarty.php]: last commit:
`$work_dir` is get from config in `FriendicaSmartyEngine` and passed to the constructor of `FriendicaSmarty`.

The call to `setUseSubDirs()` that was in `FriendicaSmartyEngine` constructor has been moved into  `FriendicaSmarty`, with all other smarty setup functions.

**Comment by MrPetovan** [comment]
Do you mind me taking over this PR to implement @nupplaphil's suggestions and fix the tests?

**Comment by fabrixxm** [comment]
go on.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
And here we go! I squashed all the fixup commits into one, moved the default config value to the `static/defaults.config.php` file, rebased onto develop and ran `sh bin/run_xgettext.sh` from my Friendica base folder.

**Comment by nupplaphil** [comment]
And comes the merge :-)

**Pull request merged**